### DISCUSSION TITLE
Using Go Shared Libraries with DLLImport on Android and iOS

### DISCUSSION THREAD

**Discussion by DanielcoderX** [AvaloniaUI/Avalonia#13141] [Q&A] [answered]
Greetings, fellow developers! I've recently achieved success in transforming my **Golang** file into a shared library, accessible through **`DLLImport`** on both Linux and Windows platforms. However, my curiosity now turns to the realms of Android and iOS.

I'm aware of the involvement of **gomobile** in the process of building libraries for mobile platforms. Yet, the compatibility of **`DLLImport`** with Android and iOS remains a question mark for me. Has anyone explored this territory? If you have experience or insights to share regarding the use of **`DLLImport`** with shared libraries on Android and iOS, your knowledge would be immensely valuable! Your hints or detailed processes would be greatly appreciated!
Thank you in advance for your contributions! 🚀

**Comment by timunie** [comment]
probably worth to check MAUI community how they handle this. They have a bigger community when it comes to mobile, Avalonia is quite new to this area.

**Reply by timunie** [reply, marked as answer]
Same approach should work using Avalonia imo.
